### ISSUE TITLE
Fix false positives for double-slash comments within selector lists in no-invalid-double-slash-comments

### ISSUE THREAD

**Issue by haggen** [stylelint/stylelint#4877]
> Clearly describe the bug

When running stylelint I'm getting a report of a violation [no-invalid-double-slash-comments](https://stylelint.io/user-guide/rules/no-invalid-double-slash-comments), which should never happen since double slash comments are supported by Less.

It seems to **only happen when I write a comment in the middle of a selector**, which also should be okay in Less land.

> Which rule, if any, is the bug related to?

- no-invalid-double-slash-comments

> What code is needed to reproduce the bug?

```css
// Class .one
.one,
// Class .two
.two {
  color: red;
}
```

> What stylelint configuration is needed to reproduce the bug?

```js
module.exports = {
  extends: "stylelint-config-standard",
};

```

> Which version of stylelint are you using?

`13.6.1`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

CLI with `npx stylelint "**/*.less`

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

Yes. Less.

> What did you expect to happen?

No violations should've been reported.

> What actually happened (e.g. what warnings or errors did you get)?

```shell
src/styles/antd.less
 2:1  ✖  Unexpected double-slash CSS comment   no-invalid-double-slash-comments
 2:1  ✖  Cannot parse selector                 parseError                      
 2:1  ✖  Cannot parse selector                 parseError                      
 2:1  ✖  Cannot parse selector                 parseError                      
 2:1  ✖  Cannot parse selector                 parseError                      
 2:1  ✖  Cannot parse selector                 parseError                      
 2:1  ✖  Cannot parse selector                 parseError                      
 2:1  ✖  Cannot parse selector                 parseError    
```

**Comment by hudochenkov** [comment]
You need to disable this rule for Less. All core rules are designed for standard CSS syntax, if rule doesn't make sense for specific syntax, user should disable it.

**Comment by haggen** [comment]
@hudochenkov I think you're mistaken because if I move the comment that's in the middle of the selector elsewhere it works. It's not like any double-slash comment breaks it. Stylelint says it supports Less, Scss, etc and it does. It's obviously an edge case bug.

**Comment by hudochenkov** [comment]
I see few problems:

1. Indeed `no-invalid-double-slash-comments` reporting is inconsistent in less files, and SCSS, for example. [Demo](https://stylelint.io/demo#N4Igxg9gJgpiBcID0SAEBhANgQwM69QDoIA7GAHROLIBpKUMd8iAXAdwksPYlWEtSpImCACd4qUTCgBuSgF8QNELgCeJFtgAeCEJhj4l4UgDMAlgHNd-EoPIhRAV3257Em4LsgSEALRmSADdsTDMoXygIRwAjfV9cJgALX0gAW1SYDVcEVBYnCltUeQUQeSA). Probably also `postcss-scss` and `postcss-less` produce different AST, which gives different output for these syntaxes.
2. Some of rules from `stylelint-config-standard` report parse errors (Cannot parse selector parseError) for Less syntax and example from the first comment. However, no parse errors, when parser is SCSS. Probably, again `postcss-less` produces different AST or our rules misbehaving. It is not known what rules throw parser errors.

**Comment by haggen** [comment]
@hudochenkov Thank you for taking another look into it. One odd thing I noticed is that the parse errors disappear once I remove the comment mixed in the selector. i.e. the following works just fine:

```less
// Class .one
// Class .two
.one,
.two {
  color: red;
}
```

I have no prior experience with parsers and/or linting tools but it seems to me that this sequence `selection, comment, selector` is throwing it off and all hell breaks loose. Perhaps of an unrecognizable AST, like you mentioned.

**Comment by malsf21** [comment]
Sounds like this may be an upstream issue with `postcss-less`, though needs some more investigation. Going to label it as such - probably warrants further investigation!

**Comment by jeddy3** [comment]
> Indeed no-invalid-double-slash-comments reporting is inconsistent in less files, and SCSS, for example. Demo. Probably also postcss-scss and postcss-less produce different AST, which gives different output for these syntaxes.

I agree. Comments within selector lists aren't being parsed as comment nodes by postcss-less.

> Some of rules from stylelint-config-standard report parse errors (Cannot parse selector parseError) for Less syntax and example from the first comment. However, no parse errors, when parser is SCSS. Probably, again postcss-less produces different AST or our rules misbehaving. It is not known what rules throw parser errors.

Yes, the "Cannot parse selector" errors are being caused by another rule, and not `no-invalid-double-slash-comments`. I suspect it's the same underlying issue in the parser.

---

@haggen Please consider [contributing to the parser](https://stylelint.io/developer-guide/syntaxes#working-on-syntaxes) if you have time.

**Comment by jeddy3** [comment]
Closing as the [`syntax` option was removed](https://github.com/stylelint/stylelint/issues/5289) for the [14.0.0 release](https://github.com/stylelint/stylelint/issues/5205). (See the [migration guide](https://github.com/stylelint/stylelint/blob/main/docs/migration-guide/to-14.md#syntax-option-and-automatic-inferral-of-syntax).)

Please consider fixing the issue upstream as the syntax is no longer bundled by Stylelint.
